File: src/atr_pkg/scripts/ma_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry 


def chao_callback(msg):
    rospy.loginfo(msg.star)
    rospy.loginfo(msg.grade)
    rospy.loginfo(msg.data)

def yao_callback(msg):
    pass



if __name__ == "__main__":
    rospy.init_node("ma_node")

    sub = rospy.Subscriber("kuai_shang_che", Carry, chao_callback,queue_size=10)
    sub2 = rospy.Subscriber("ge_ge_dai_wo", Carry, yao_callback,queue_size=10)
    

    rospy.spin()

File: ch3/src/wzry_pkg/src/ma_node.cpp
#include "ros/ros.h"
#include "qq_msgs/Carry.h"
#include "std_msgs/String.h"


void CarryCallback(const qq_msgs::Carry msg)
{

    ROS_INFO("data:%s\tgrade:%s\tstar:%d",msg.data.c_str(),msg.grade.c_str(),msg.star);
}


int main(int argc, char *argv[])
{
    ros::init(argc, argv, "ma_node");

    ros::NodeHandle n;
    ros::Subscriber sub = n.subscribe<qq_msgs::Carry>("wang_zhe_kai_hei_qun", 10, &CarryCallback);

    ros::spin();
    

    return 0;
}
